mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.java

```diff
@@ -662,6 +663,23 @@ private RecordedRequest readRequest(Socket socket, BufferedSource source, Buffer
       sink.flush();
     }
 
+    if (socketPolicy == MULTIPLE_1XX) {
+      sink.writeUtf8("HTTP/1.1 100 Continue\r\n");
+      sink.writeUtf8("Content-Length: 0\r\n");
+      sink.writeUtf8("\r\n");
+      sink.flush();
+
+      sink.writeUtf8("HTTP/1.1 101 Switching Protocols\r\n");
```

Comment:
I don’t think we should proceed to do HTTP after receiving an HTTP 101.

Instead should we just explicitly whitelist 100 and 102?

Reply:
(The check for websocket support in particular makes me anxious.)